internal/provider/data_source_test.go

```diff
@@ -11,67 +11,19 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
-type TestHttpMock struct {
-	server *httptest.Server
-}
-
-const testDataSourceConfig_basic = `
-data "http" "http_test" {
-  url = "%s/meta_%d.txt"
-}
-
-output "body" {
-  value = data.http.http_test.body
-}
-
-output "response_headers" {
-  value = data.http.http_test.response_headers
-}
-`
-
 func TestDataSource_http200(t *testing.T) {
 	testHttpMock := setUpMockHttpServer()
 
 	defer testHttpMock.server.Close()
 
 	resource.UnitTest(t, resource.TestCase{
-		Providers: testProviders,
+		ProviderFactories: testProviders(),
 		Steps: []resource.TestStep{
 			{
-				Config: fmt.Sprintf(testDataSourceConfig_basic, testHttpMock.server.URL, 200),
-				Check: func(s *terraform.State) error {
-					_, ok := s.RootModule().Resources["data.http.http_test"]
-					if !ok {
-						return fmt.Errorf("missing data resource")
-					}
-
-					outputs := s.RootModule().Outputs
-
-					if outputs["body"].Value != "1.0.0" {
-						return fmt.Errorf(
-							`'body' output is %s; want '1.0.0'`,
-							outputs["body"].Value,
-						)
-					}
-
-					response_headers := outputs["response_headers"].Value.(map[string]interface{})
-
-					if response_headers["X-Single"].(string) != "foobar" {
-						return fmt.Errorf(
-							`'X-Single' response header is %s; want 'foobar'`,
-							response_headers["X-Single"].(string),
-						)
-					}
-
-					if response_headers["X-Double"].(string) != "1, 2" {
-						return fmt.Errorf(
-							`'X-Double' response header is %s; want '1, 2'`,
-							response_headers["X-Double"].(string),
-						)
-					}
-
-					return nil
-				},
+				Config: fmt.Sprintf(testDataSourceConfigBasic, testHttpMock.server.URL, 200),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataSourceBasic("data.http.http_test"),
```

Comment:
Seems like this could be written as the below using the acceptance testing helper functions, rather than introducing indirection with a separate function:

```go
resource.TestCheckResourceAttr("data.http.http_test", "body", "1.0.0"),
resource.TestCheckResourceAttr("data.http.http_test", "response_headers.X-Single", "foobar"),
resource.TestCheckResourceAttr("data.http.http_test", "response_headers.X-Double", "1, 2"),
```

Reply:
Good point. Fixed